pkg/base/variadic_test.go

```diff
@@ -0,0 +1,330 @@
+								"spec": map[string]interface{}{
+									"containers": []interface{}{
+										map[string]interface{}{
+											"name": "hello",
+										},
+									},
+								},
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "spec",
+					Type:     "map",
+					Index:    0,
+					Data: map[string]interface{}{
+						"template": map[string]interface{}{
+							"spec": map[string]interface{}{
+								"containers": []interface{}{
+									map[string]interface{}{
+										"name": "hello",
+									},
+								},
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "template",
+					Type:     "map",
+					Index:    0,
+					Data: map[string]interface{}{
+						"spec": map[string]interface{}{
+							"containers": []interface{}{
+								map[string]interface{}{
+									"name": "hello",
+								},
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "spec",
+					Type:     "map",
+					Index:    0,
+					Data: map[string]interface{}{
+						"containers": []interface{}{
+							map[string]interface{}{
+								"name": "hello",
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "containers",
+					Type:     "array",
+					Index:    0,
+					Data: map[string]interface{}{
+						"name": "hello",
+					},
+					Array: []interface{}{
+						map[string]interface{}{
+							"name": "hello",
+						},
+					},
+				},
+			},
+			wantyaml: map[string]interface{}{
+				"spec": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"containers": []interface{}{
+								map[string]interface{}{
+									"name": "hello",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			req := require.New(t)
+
+			result := buildYamlFromStack(test.stack)
+
+			req.Equal(test.wantyaml, result)
+		})
+	}
+}
+
+func Test_generateTargetValue(t *testing.T) {
+	tests := []struct {
+		name             string
+		configOptionName string
+		valueName        string
+		target           string
+		want             interface{}
+	}{
+		{
+			configOptionName: "secret",
+			valueName:        "secret-1",
+			target:           "repl{{ ConfigOption \"secret\" }}",
+			want:             "repl{{ ConfigOption \"secret-1\" }}",
+		},
+		{
+			configOptionName: "secret",
+			valueName:        "secret-1",
+			target:           "repl{{ ConfigOptionName \"secret\" }}",
+			want:             "repl{{ ConfigOptionName \"secret-1\" }}",
+		},
+		{
+			configOptionName: "secret",
+			valueName:        "secret-1",
+			target:           "repl{{ ConfigOptionFilename \"secret\" }}",
+			want:             "repl{{ ConfigOptionFilename \"secret-1\" }}",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			req := require.New(t)
+			result := generateTargetValue(test.configOptionName, test.valueName, test.target)
```

Comment:
I think this function is very unstable.  Also templates can be quoted in a variety of ways.  Some examples:
```
value: repl{{ ConfigOptionFilename "secret" }}
value: "repl{{ ConfigOptionFilename "secret" }}"
value: "repl{{ ConfigOptionFilename 'secret' }}"
value: "repl{{ ConfigOptionFilename `secret` }}
```

I think we should explore other options, for example, another template can be used

```
value: "repl{{ ConfigOptionFilename 'secret-[[repl i]]' }}"
```

Reply:
Fixed with subtemplating, as discussed.